### PR TITLE
parquet pipeline

### PR DESCRIPTION
**Pull request by jpn--**
Initial implementation for #645

### CODE REVIEW

File: activitysim/core/pipeline.py [hunk 1]
```diff
@@ -99,7 +100,11 @@ def close_open_files():
 
 def open_pipeline_store(overwrite=False, mode="a"):
     """
-    Open the pipeline checkpoint store
+    Open the pipeline checkpoint store.
+
+    If the pipeline_file_name setting ends in ".h5", then the pandas
+    HDFStore file format is used, otherwise pipeline files are stored
+    as parquet files organized in regular file system directories.
```
Comment (dhensle) [inline activitysim/core/pipeline.py:107]: I assume documentation on this setting will to be addressed in the other Pydantic task?

File: activitysim/core/pipeline.py [hunk 2]
```diff
@@ -193,7 +216,11 @@ def write_df(df, table_name, checkpoint_name=None):
     We store multiple versions of all simulation tables, for every checkpoint in which they change,
     so we need to know both the table_name and the checkpoint_name to label the saved table
 
-    The only exception is the checkpoints dataframe, which just has a table_name
+    The only exception is the checkpoints dataframe, which just has a table_name,
+    although when using the parquet storage format this file is stored as "None.parquet"
+    to maintain a simple consistent file directory structure.
+
+    If the
```
Comment (dhensle) [inline activitysim/core/pipeline.py:223]: Looks like unfinished thought here...

File: activitysim/core/pipeline.py [hunk 3]
```diff
@@ -209,10 +236,28 @@ def write_df(df, table_name, checkpoint_name=None):
...
     store = get_pipeline_store()
-
-    store[pipeline_table_key(table_name, checkpoint_name)] = df
-
-    store.flush()
+    if isinstance(store, Path):
+        store.joinpath(table_name).mkdir(parents=True, exist_ok=True)
+        df.to_parquet(store.joinpath(table_name, f"{checkpoint_name}.parquet"))
+    else:
+        complib = config.setting("pipeline_complib", None)
```
Comment (dhensle) [inline activitysim/core/pipeline.py:243]: Another setting to not get lost in the Pydantic task.

**Review by dhensle — COMMENTED** [review]
Generally looks good and seems to work as designed for me.  Wondering if the default pipeline_file_name setting should change in config.py to remove the ".h5". (Had to do some digging here for the actual setting to get this feature turned on!).  Note that I think changing the default setting will also use this parquet functionality in the pipeline CI testing, which I think is desirable.  Based on the getting_started.ipynb notebooks, I think the .h5 version should be specified explicitly in the settings for the prototype_mtc example.

**Comment by jpn--** [comment]
This PR was superseded by #654

**Pull request closed without merging**